### PR TITLE
Add curated plugin doc and collection fields

### PR DESCRIPTION
**Pull request by mfridman**

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/plugin_curation.proto [hunk 1]
```diff
@@ -112,6 +112,21 @@ enum DotnetTargetFramework {
   DOTNET_TARGET_FRAMEWORK_NET_8_0 = 13;
 }
 
+// The collection a Plugin belongs to. These collections help organize plugins based on their
+// functionality or ecosystem.
+enum PluginCollection {
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:117]: This feels like a completely new entity, not an enum - it should probably be dynamic

Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto]: Sure, updated. I could see us expanding this to include a description, maybe a logo, create_time, etc.

Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto]: This doesn't appear to be an entity - it has no ID as an example.

How does this fit into the new API, why is this going here, etc? Need more details.

Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto]: It's a hard-coded list of plugin mappings to collections. These aren't intended to be user-configurable and there's no mechanism for users to edit these collections (at least for now).

The goal is to display older protoc plugins in the UI under these collections.

This is something we should put more careful thought into when designing the newer `plugin.v1` API in registry-proto, but to unblock certain UI work feels fine to piggyback on the existing primitives?

**Review by bufdev — APPROVED** [review]
Approving to unblock, but this shape won't be approved in the API that we will be using in v1

**Pull request merged**